catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -44,7 +54,47 @@ public Partition[] listPartitions() {
 
   @Override
   public Partition getPartition(String partitionName) throws NoSuchPartitionException {
-    throw new UnsupportedOperationException();
+    try {
+      org.apache.hadoop.hive.metastore.api.Partition partition =
+          table
+              .clientPool()
+              .run(c -> c.getPartition(table.schemaName(), table.name(), partitionName));
+      return fromHivePartition(partitionName, partition);
+
+    } catch (UnknownTableException e) {
+      throw new NoSuchTableException(
+          "Hive table " + table.name() + " does not exist in Hive Metastore", e);
+
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchPartitionException(
+          "Hive partition " + partitionName + " does not exist in Hive Metastore", e);
+
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(
+          "Failed to get partition "
+              + partitionName
+              + " of table "
+              + table.name()
+              + "from Hive Metastore",
+          e);
+    }
+  }
+
+  private Partition fromHivePartition(
+      String partitionName, org.apache.hadoop.hive.metastore.api.Partition partition) {
+    String[][] fieldNames = getFieldNames(partitionName);
+    Literal[] values =
+        partition.getValues().stream().map(Literals::stringLiteral).toArray(Literal[]::new);
+    // todo: support partition properties metadata to get more necessary information
```

Comment:
What is the TODO here?

Reply:
Here are some partition metadata such as `location` and `SerdeInfo` that we have not handled, so I leave a todo to deal with in the next version.